code/obj/item/device/pda2/pda2.dm

```diff
@@ -423,12 +423,18 @@
 		if (U.try_deliver(C, user))
 			return
 
-	if (istype(C, /obj/item/disk/data/cartridge) && isnull(src.cartridge))
+	if (istype(C, /obj/item/disk/data/cartridge))
 		user.drop_item()
 		C.set_loc(src)
-		boutput(user, "<span class='notice'>You insert [C] into [src].</span>")
-		src.cartridge = C
-		src.updateSelfDialog()
+		if (isnull(src.cartridge))
+			boutput(user, "<span class='notice'>You insert [C] into [src].</span>")
+			src.cartridge = C
+			src.updateSelfDialog()
+		else
+			boutput(user, "<span class='notice'>You remove the old cartridge and insert [C] into [src].</span>")
+			user.put_in_hand_or_eject(src.cartridge)
+			src.cartridge = C
+			src.updateSelfDialog()
```

Comment:
could bring these two lines out of the if/else, since it'll be true no matter what

Reply:
good eye. fixed